project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/sample/qualitycontrol/QualityControlStore.java

```diff
@@ -0,0 +1,105 @@
+
+import static org.apache.logging.log4j.LogManager.getLogger;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import life.qbic.application.commons.ApplicationException;
+import life.qbic.projectmanagement.application.api.PurchaseStoreException;
+import life.qbic.projectmanagement.application.sample.qualitycontrol.QualityControlStorage;
+import life.qbic.projectmanagement.application.sample.qualitycontrol.QualityControlStorageException;
+import life.qbic.projectmanagement.domain.model.project.ProjectId;
+import life.qbic.projectmanagement.domain.model.sample.qualitycontrol.QualityControl;
+import life.qbic.projectmanagement.domain.model.sample.qualitycontrol.QualityControlUpload;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * <b>Quality Control Store</b>
+ *
+ * <p>Implementation of the {@link QualityControlStorage} interface.</p>
+ */
+@Component
+public class QualityControlStore implements QualityControlStorage {
+
+  private static final Logger log = getLogger(QualityControlStore.class);
+
+  private final QualityControlJpa persistenceStore;
+
+  @Autowired
+  public QualityControlStore(QualityControlJpa persistenceStore) {
+    this.persistenceStore = persistenceStore;
+  }
+
+  /**
+   * Stores a {@link QualityControl} item persistently.
+   *
+   * @param qualityControl the QualityControlUpload item to store
+   * @throws QualityControlStorageException is thrown if the storage fails
+   * @since 1.0.0
+   */
+  @Override
+  public void storeQualityControl(QualityControl qualityControl)
+      throws QualityControlStorageException {
+    try {
+      persistenceStore.save(qualityControl);
+    } catch (RuntimeException e) {
+      throw new PurchaseStoreException(
+          "Storing the quality control for project %s failed" .formatted(qualityControl.project()),
+          e);
+    }
+  }
+
+  @Override
+  public void storeQualityControls(List<QualityControl> qualityControls)
+      throws QualityControlStorageException {
+    try {
+      persistenceStore.saveAll(qualityControls);
+    } catch (RuntimeException e) {
+      throw new PurchaseStoreException("Storing quality controls failed.");
+    }
+  }
+
+  @Override
+  public List<QualityControlUpload> findQualityControlsForProject(ProjectId projectId) {
+    try {
+      return persistenceStore.findQualityControlAssociationByProjectIdEquals(projectId).stream()
+          .sorted(Comparator.comparing(QualityControl::providedOn)) //ensures same ordering
+          .map(QualityControl::qualityControlUpload).toList();
+    } catch (RuntimeException e) {
+      throw new ApplicationException(
+          "Retrieving quality controls for project %s failed." .formatted(projectId), e);
+    }
+  }
+
+  @Override
+  public void deleteQualityControlsForProject(String projectId, long qualityControlId) {
```

Comment:
this is just for deleting a single file, right? the method name should reflect that

Reply:
Top spot 👀